www/src/components/auth/AuthButton.tsx

```diff
@@ -0,0 +1,39 @@
+import { useAuth0 } from '@auth0/auth0-react';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+type BaseButtonProps = {
+    text: string;
+    onClick: () => void;
+};
+
+const BaseButton: React.FC<BaseButtonProps> = (props) => {
+    return <button onClick={props.onClick}>{props.text}</button>;
+};
+
+const LogoutButton = () => {
+    const { logout } = useAuth0();
+    return <BaseButton text="Logout" onClick={() => logout()} />;
+};
+
+const LoginButton = () => {
+    const { loginWithRedirect } = useAuth0();
+    return <BaseButton text="Login" onClick={() => loginWithRedirect()} />;
+};
+
+const AuthButton: React.FC = () => {
+    const { isAuthenticated } = useAuth0();
+
+    if (isAuthenticated) {
+        return <LogoutButton />;
+    }
+
+    return <LoginButton />;
+};
+
+BaseButton.propTypes = {
```

Comment:
Is this required? I ask because I've never seen this tbh.

Reply:
I was able to get rid of this in this file by not using `React.FC` but this restricts us from making components that can render a `children` prop. Ideally I'd just disable this setting as mentioned in [React eslint error missing in props validation](https://stackoverflow.com/questions/38684925/react-eslint-error-missing-in-props-validation) what do you think?
![image](https://user-images.githubusercontent.com/43416725/120737853-13c48f80-c4ac-11eb-9f15-60b2ad8d590e.png)

Reply:
Was able to fix it, I thought typescript could infer the types from `FC` directly but seems like it needs us to explicitly type the props in the function itself. Thanks for showing me this 🍻